### ISSUE TITLE
Implement grouping options for duration type

### ISSUE THREAD

**Issue by kgodey** [mathesar-foundation/mathesar#432]
## Problem
We need to ensure that records that include columns of duration types support range based grouping just like Number types via the API. Please see #384 for the implementation details. 

We should be able to reuse the implementation for Number types here.

This involves:
- Implementing the grouping in the backend
- Updating the `/api/v0/databases/<id>/types/` endpoint to store available grouping on this type

## Additional context
- Please see these PRs for code related to grouping:
  - #315
  - #353
- This issue is blocked by implementation of:
  - #430  
  - #384
- #557 provides some context on storing groups.

**Comment by shivansii** [comment]
hey! can I work on this issue?

**Comment by rajatvijay** [comment]
@shivansii Sure, but note that since this does not have a `help-wanted` or `good-first` label, has a higher difficulty level.

**Comment by seancolsen** [comment]
Unassigning due to inactivity

**Comment by kgodey** [comment]
Closing this, too old, spec is likely wrong. We can create a new issue if we need this functionality in the future.
